### PR TITLE
bpf: Store post-preprocessor C for endpoint

### PR DESCRIPTION
**Pull request by joestringer**
For debugging purposes, it is useful to have the preprocessed C for the
datapath available in `/var/run/cilium/state/<EP>/`.

Signed-off-by: Joe Stringer <joe@covalent.io>

### CODE REVIEW

**Review by aalemayhu — APPROVED** [review]

**Review by aanm — APPROVED** [review]

File: bpf/join_ep.sh [hunk 1]
```diff
@@ -28,6 +28,9 @@ function bpf_compile()
 	OUT=$2
 	TYPE=$3
 
+	clang -E -O2 -target bpf					\
```
Comment (tgraf) [inline bpf/join_ep.sh]: How long does this take? We will pay this cost every time we build endpoints. Can we make it dependant on the debug flag?

**Review by tgraf — DISMISSED** [review]

File: bpf/join_ep.sh (hunk 1 above)
Comment (joestringer) [reply bpf/join_ep.sh]: Ack, making that change.

**Review by jrfastab — APPROVED** [review]
Sure in debug mode seems OK. Just lxc though?

**Comment by joestringer** [comment]
There's no debug mode for the other progs. Could add it and do the same for those.

**Pull request merged**